src/pages/suggestions/index.tsx

```diff
@@ -39,30 +48,80 @@ const Suggestions: FC = () => {
       })
   }, [supabaseClient])
 
+  useEffect(() => {
+    if (showSearchBar && inputRef.current) {
+      inputRef.current.focus()
+    }
+  }, [showSearchBar])
+
+  const handleSearch = (e: ChangeEvent<HTMLInputElement>) => {
+    const searchText: string = e.target.value
+
+    const filteredSuggestions = suggestions.filter((suggestion) => {
+      const { title, motivation, artist } = suggestion
+      const lowerCaseSearchText = searchText.toLowerCase()
+      return (
+        title.toLowerCase().includes(lowerCaseSearchText) ||
+        motivation.toLowerCase().includes(lowerCaseSearchText) ||
+        artist.some((artist) => artist.toLowerCase().includes(lowerCaseSearchText))
+      )
+    })
+
+    if (filteredSuggestions.length === 0) {
+      setNoSuggestionsText(
+        "Looks like the song you are looking for does not exist yet! Feel free to add the song!"
+      )
+    } else {
+      setNoSuggestionsText("")
+    }
+
+    setSearchedSuggestions(filteredSuggestions)
+  }
+
+  const renderSuggestions = () => {
+    const displayedSuggestions = showSearchBar ? searchedSuggestions : suggestions
+
+    return (
+      <div className={"flex flex-wrap justify-center gap-6"} data-cy="suggestions-list">
+        {displayedSuggestions.map((suggestion) => (
+          <SuggestionCard key={suggestion.id} suggestion={suggestion} />
+        ))}
+      </div>
+    )
+  }
+
   return (
     <div className={"page-wrapper"}>
-      <div className={"flex justify-between"}>
+      <div className={"flex justify-between"} style={{ display: showSearchBar ? "none" : "flex" }}>
         <div className={"page-header"}>Suggestions</div>
-        <PlusIcon
-          data-cy={"button-new-suggestion"}
-          className={"h-8 w-8 cursor-pointer text-black hover:text-zinc-400"}
-          onClick={() => router.push("/suggestions/new")}
-        />
+        <div className={"flex flex-row gap-2"}>
+          <MagnifyingGlassIcon
+            data-cy={"button-search-suggestions"}
```

Comment:
It's a subtle difference but could you add this line of code?
```css
viewBox={"-1.25 -1.25 23.25 23.25"}
```

It makes the icons the same size.

Without the line:
![image](https://github.com/ASVGay/the-rhapsodies/assets/54884540/077f7549-0090-4528-b247-73190bae4fd0)
With the line:
![image](https://github.com/ASVGay/the-rhapsodies/assets/54884540/f3bae5be-e54c-44bf-a78d-0bb5b3a033a2)

Reply:
@maarcelh have you added this yet?

Reply:
seems i forgot that one, added it!